File: backend/src/main/java/com/iset/covtn/models/Rating.java
package com.iset.covtn.models;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Rating {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String comment;
    private int rating;

    @JsonIgnoreProperties({
            "ratings",
            "ratingsSent",
            "car",
            "rideParticipations"
    })
    @ManyToOne
    private UserInfo user;

    @JsonIgnoreProperties({
            "ratings",
            "ratingsSent",
            "car",
            "rideParticipations"
    })
    @ManyToOne
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private UserInfo targetUser;

}

File: mobile/CovTn/app/src/main/java/com/iset/covtn/models/Rating.kt
package com.iset.covtn.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Rating (
    val id: Long,
    val comment: String,
    val rating : Int,
    val user: User,

    val targetUser: User? = null
): Parcelable